src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,69 @@
+
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo | null,
+  setSelectedTodo: (value: Todo | null) => void,
+}
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(Number(selectedTodo?.userId))
+      .then(setUser);
+  }, [selectedTodo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {user && selectedTodo
+        ? (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                Todo #
+                {selectedTodo.id}
+              </div>
+
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => setSelectedTodo(null)}
+              />
+            </header>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {selectedTodo.title}
+              </p>
+
+              <p className="block" data-cy="modal-user">
+                {selectedTodo.completed
+                  ? <strong className="has-text-success">Done</strong>
+                  : <strong className="has-text-danger">Planned</strong>}
+                {' by '}
+                <a href={user.email}>
+                  {user.name}
+                </a>
+              </p>
```

Comment:
fix it that way that on click user has the ability to send email
![image](https://user-images.githubusercontent.com/59209337/213642468-1f57563d-bb49-41d9-bd44-6bd658bc0904.png)